thewalrus/libwalrus.pyx

```diff
@@ -468,5 +546,229 @@ def renorm_hermite_multidimensional_real(double [:, :] R, double [:] y, int cuto
+    r"""Returns the matrix elements of a displacement by using a custom version of
+    the renormalized Hermite polynomials.
+
+    Args:
+        y (array[float64]): vector parametrizing the displacement. It has the form :math:`[x, -x]`
+        cutoff (int): maximum size of the subindices in the tensor
+
+    Returns:
+        array[float64]: the matrix elements of the real displacement operator
+    """
+    cdef int i, n = 2
+    cdef vector[double] y_mat
+    for i in range(n):
+        y_mat.push_back(y[i])
+    length = cutoff**n
+    cdef double *array = displacement_cpp(y_mat, cutoff)
+    cdef np.ndarray ndarray
+    array_wrapper = ArrayWrapperFloat()
+    array_wrapper.set_data(length, <void*> array)
+    ndarray = np.array(array_wrapper, copy=False)
+    ndarray.base = <PyObject*> array_wrapper
+    Py_INCREF(array_wrapper)
+    return np.reshape(ndarray, [cutoff]*n)
+
+def two_mode_squeezing(double complex[:, :] R, int cutoff):
+    r"""Returns the matrix elements of a two mode squeezer by using a custom version of
+    the renormalized Hermite polynomials.
+
+    Args:
+        R (array[complex128]): square matrix parametrizing the two-mode squeezing operator
+        cutoff (int): maximum size of the subindices in the tensor
+
+    Returns:
+        array[complex128]: the matrix elements of the two-mode squeezing operator
+    """
+    cdef int i, j, n = R.shape[0]
+    cdef vector[double complex] R_mat
+
+    for i in range(n):
+        for j in range(n):
+            R_mat.push_back(R[i, j])
+    length = cutoff**n
+    cdef double complex *array = two_mode_squeezing_cpp(R_mat, cutoff)
+    cdef np.ndarray ndarray
+    array_wrapper = ArrayWrapper()
+    array_wrapper.set_data(length, <void*> array)
+    ndarray = np.array(array_wrapper, copy=False)
+    ndarray.base = <PyObject*> array_wrapper
+    Py_INCREF(array_wrapper)
+    return np.reshape(ndarray, [cutoff]*n)
+
+def two_mode_squeezing_real(double [:, :] R, int cutoff):
+    r"""Returns the matrix elements of a single mode squeezer by using a custom version of
+    the renormalized Hermite polynomials.
+
+    Args:
+        R (array[float64]): square matrix parametrizing the two-mode squeezing operator
+        cutoff (int): maximum size of the subindices in the tensor
+
+    Returns:
+        array[float64]: the matrix elements of the two-mode squeezing operator
+    """
+    cdef int i, j, n = R.shape[0]
+    cdef vector[double] R_mat
 
-    return renorm_hermite_multidimensional_cpp(R_mat, y_mat, cutoff)
+    for i in range(n):
+        for j in range(n):
+            R_mat.push_back(R[i, j])
+    length = cutoff**n
+    cdef double *array = two_mode_squeezing_cpp(R_mat, cutoff)
+    cdef np.ndarray ndarray
+    array_wrapper = ArrayWrapperFloat()
+    array_wrapper.set_data(length, <void*> array)
+    ndarray = np.array(array_wrapper, copy=False)
+    ndarray.base = <PyObject*> array_wrapper
+    Py_INCREF(array_wrapper)

```

Comment:
Since the above code is repeated for every function, is it possible to make it it's own function, and so it's only a one line call here?